<a href="https://colab.research.google.com/github/abdimare/DS-Unit-1-Sprint-1-Data-Wrangling-and-Storytelling/blob/master/Abdrehim_NuruCopy_of_LS_DS_121_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 2 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

In [2]:
### YOUR CODE STARTS HERE
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data

--2020-04-14 12:12:24--  https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data
Resolving archive.ics.uci.edu (archive.ics.uci.edu)... 128.195.10.252
Connecting to archive.ics.uci.edu (archive.ics.uci.edu)|128.195.10.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 18171 (18K) [application/x-httpd-php]
Saving to: ‘house-votes-84.data’

house-votes-84.data 100%[===================>]  17.75K  --.-KB/s    in 0.1s    

2020-04-14 12:12:25 (137 KB/s) - ‘house-votes-84.data’ saved [18171/18171]



In [3]:
import pandas as pd

column_headers = ['party','handicapped-infants','water-project',
                  'budget','physician-fee-freeze', 'el-salvador-aid',
                  'religious-groups','anti-satellite-ban',
                  'aid-to-contras','mx-missile','immigration',
                  'synfuels', 'education', 'right-to-sue','crime','duty-free',
                  'south-africa']

df = pd.read_csv('house-votes-84.data', names=column_headers)

print(df.shape)
df.head()

(435, 17)


party handicapped-infants water-project  ... crime duty-free south-africa
0  republican                   n             y  ...     y         n            y
1  republican                   n             y  ...     y         n            ?
2    democrat                   ?             y  ...     y         n            n
3    democrat                   n             y  ...     n         n            y
4    democrat                   y             y  ...     y         y            y

[5 rows x 17 columns]

In [4]:
# Categorical encoding: replace strings that represent specific categories with integers.
import numpy as np

df = df.replace({'y': 1, 'n': 0, '?': np.NaN})

df.head()

party  handicapped-infants  ...  duty-free  south-africa
0  republican                  0.0  ...        0.0           1.0
1  republican                  0.0  ...        0.0           NaN
2    democrat                  NaN  ...        0.0           0.0
3    democrat                  0.0  ...        0.0           1.0
4    democrat                  1.0  ...        1.0           1.0

[5 rows x 17 columns]

In [5]:
df['party'].value_counts()

democrat      267
republican    168
Name: party, dtype: int64

In [0]:
dem = df[df['party']=='democrat']
rep = df[df['party']=='republican']

In [19]:
#I am going to test the hyppthesis that says democrats supports the 'immigration bill' more tha republican
dem['immigration'].mean()


0.4714828897338403

In [18]:
rep['immigration'].mean()


0.5575757575757576

#define the null and alternate hyphothesis
#Null hyphosis:Demcrats supports the 'immigration bill' equally republican/50%-50%/
#Alteranate hypthesis:Democrats supports the the immigation bill more than  republican 
#Confidence interval=99%(0.01 P-value thereshold)

In [9]:
from scipy import stats
stats.ttest_ind(dem['immigration'], rep['immigration'], nan_policy='omit')

Ttest_indResult(statistic=-1.7359117329695164, pvalue=0.08330248490425066)

The T-stastics is-1.7359117 and the p-value is 0.0833
If the P-value is less than (or equal to) alpha/significance level/, then the null hypothesis is rejected in favor of the alternative hypothesis. And, if the P-value is greater than , then the null hypothesis is not rejected.
So,Based on a t-statisc of -1.7359117 and a p-value of .0083 I fail to reject the null hypothesis that says democrats spports the immgration bill more than republican.The negative value of T-statistics indicates that the mean of republicans is greater than the mean of democrats

In [0]:
#I am going to test the hyppthesis that says republican supports the 'budget bill' more tha democrats
# Null hyphosis=Republican supports the 'budget bill' equally as democrats did /50%-50%
#Alternate hyphothesis=both parties supports the budjet bill in different proportion
#Confidence interval=99%/0.01 P-value thereshold/

In [14]:
rep['budget'].mean()

0.13414634146341464

In [16]:
rep['budget'].mean()

0.13414634146341464

In [21]:
stats.ttest_ind(dem['budget'], rep['budget'], nan_policy='omit')

Ttest_indResult(statistic=23.21277691701378, pvalue=2.0703402795404463e-77)

The T-stastics is 23.2127769 and the p-value is 2.07034 
If the P-value is less than (or equal to) alpha/significance level/, then the null hypothesis is rejected in favor of the alternative hypothesis. 
if the P-value is greater than , then the null hypothesis is not rejected.
 So,Based on a t-statisc of 23.2127769 and a p-value of 2.07034 I fail to reject the null hypothesis that says republican spports the budget bill more than democrats.Furthermore,the equality of the two means shows that i have to accept the null hypothesis

## Stretch Goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Work on Performing a T-test without using Scipy in order to get "under the hood" and learn more thoroughly about this topic.
### Start with a 1-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://blog.minitab.com/hs-fs/hubfs/Imported_Blog_Media/701f9c0efa98a38fb397f3c3ec459b66.png?width=247&height=172&name=701f9c0efa98a38fb397f3c3ec459b66.png) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then try a 2-sample t-test
 - Establish the conditions for your test 
 - [Calculate the T Statistic](https://lh3.googleusercontent.com/proxy/rJJ5ZOL9ZDvKOOeBihXoZDgfk7uv1YsRzSQ1Tc10RX-r2HrRpRLVqlE9CWX23csYQXcTniFwlBg3H-qR8MKJPBGnjwndqlhDX3JxoDE5Yg) (You'll need to omit NaN values from your sample).
 - Translate that t-statistic into a P-value. You can use a [table](https://www.google.com/search?q=t+statistic+table) or the [University of Iowa Applet](https://homepage.divms.uiowa.edu/~mbognar/applets/t.html)

 ### Then check your Answers using Scipy!